### PR TITLE
Added ability to track and untrack goals

### PR DESCRIPTION
**Pull request by carsonwolber**
## Description

- This PR adds the styling and components for showing both the general list of "generic" goals as well as a list of active goals the user has subscribed to
- Users can track and stop stacking goals

- The next PR will set up notifications, notifications are called when a user adds/removes goals and on transaction sync endpoint to be in sync with the rest of the project

## Milestones
This works on TC 1, specifically 
- EconMetrics offers the ability to set and track goals. As those goals are met / missed, users are reminded of their goals.

## Resources
The general flow of the goal components was directly inspired by the work I did in codepath (I basically did the week 3 codepath project with more complex database handling in this PR) 

## Test Plan
This video shows off what tracking/untracking goals looks like. It's pretty basic UI-wise but the focus now is on personalization

https://github.com/user-attachments/assets/5b55076b-978d-4976-85cb-c249c5304d38

### CODE REVIEW

File: frontend/src/GoalPage/AddableGoal.jsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+function AddableGoal({goal, setActiveGoals}) {
+
+    const { userId } = useParams();
+
+    const message = goalFormatter(goal.category, goal.target);
+
+    const addGoal = async() => {
+        try { 
+            const _ = followNewGoal(goal.id, userId)
```
Comment (William-La) [inline frontend/src/GoalPage/AddableGoal.jsx]: followNewGoal doesn't seem to return anything, so there is no need to assign its return value to anything (even if it's an empty underscore var).

File: frontend/HelperFuncs/utils.js [hunk 2]
```diff
@@ -475,6 +452,115 @@ export const findGenericGoals = async(transactionData) => {
...
+    message += category + ' by ' + target.toString() + '%'
+    return message
+}
+
 
+/* 
+helper to append the requested goal onto the users Goals column and update
+the corresponding goal entry to include user as a tracked user
+*/
+export const followNewGoal = async(goalId, userId) => {
```
Comment (William-La) [inline frontend/HelperFuncs/utils.js]: is there a need for followNewGoal to be a helper? It's only used once in another file. I'm all for helper functions, but sometimes it's nicer to keep logic that relates to a certain feature in one file, especially if that logic isn't being used elsewhere. We could also define the helper function in the file it's being used in-- if it is not going to be used in other files, no need to move it into utils.

Comment (carsonwolber) [reply frontend/HelperFuncs/utils.js]: Yeah I don't see a use case for this endpoint outside the AddableGoal container so I can go ahead and move all endpoint logic in there if that would be cleaner

File: frontend/src/GoalPage/AddableGoal.jsx (hunk 1 above)
Comment (carsonwolber) [reply frontend/src/GoalPage/AddableGoal.jsx]: addressed in your other  review, if I'm removing the helper this line is obsolete

File: frontend/HelperFuncs/utils.js (hunk 2 above)
Comment (carsonwolber) [reply frontend/HelperFuncs/utils.js]: refactored this and the sibling remove endpoint in 6714dd7

**Review by William-La — APPROVED** [review]

**Pull request merged**